### PR TITLE
Use datatable column descriptions from server-side

### PR DESCRIPTION
**Pull request by b-rowan**
A continuation of the server-side column handling introduced in #170, handling for the devices page is now slightly cleaner, and handling has been introduced for the remaining pages.

### CODE REVIEW

**Review by tsagadar — CHANGES_REQUESTED** [review]
Great to have this functionality unified in all screens. It comes with an additional request to show the UI (vs. embedding a column config already during template generation) but this should not really matter.

File: goosebit/ui/static/js/software.js [hunk 1]
```diff
@@ -6,6 +6,128 @@ const uploadProgressBar = document.getElementById("upload-progress");
...
+    const buttons = [
+        {
+            text: '<i class="bi bi-cloud-download" ></i>',
+            action: (e, dt) => {
+                const selectedSoftware = dt
+                    .rows({ selected: true })
+                    .data()
+                    .toArray()
+                    .map((d) => d.id);
+                downloadSoftware(selectedSoftware[0]);
```
[the remark this answers is not in the document]
Comment (b-rowan) [reply goosebit/ui/static/js/software.js:50]: This is not doing what you think.  `selectedSoftware` is an array of objects representing the software in the `SoftwareSchema` format, which is then mapped to be an array of the `id` value.  This is grabbing the first software instance in that array (which in this case is just the id) to download.

[the remark this answers is not in the document]
Comment (b-rowan) [reply goosebit/ui/static/js/software.js:50]: Hence the line 
```
                    .map((d) => d.id);
```

[the remark this answers is not in the document]
Comment (tsagadar) [reply goosebit/ui/static/js/software.js:50]: Correct - sorry for that.

**Review by tsagadar — APPROVED** [review]

**Pull request merged**